File: LeetCode/0819-most-common-word/0819-most-common-word.java
class Solution {
    public String mostCommonWord(String paragraph, String[] banned) {
        Set<String> ban = new HashSet<>(Arrays.asList(banned));
        Map<String, Integer> map = new HashMap<>();
        String[] modified = paragraph.toLowerCase()
            .replaceAll("\\W+", " ")
            .split(" ");
        
        Arrays.stream(modified)
            .filter(word -> !ban.contains(word))
            .forEach(word -> map.put(word, map.getOrDefault(word, 0) + 1));
        
        return Collections.max(map.entrySet(), Map.Entry.comparingByValue()).getKey();
    }
}

File: LeetCode/0819-most-common-word/0819-most-common-word.kt
class Solution {
    fun mostCommonWord(paragraph: String, banned: Array<String>): String? {
        val map: MutableMap<String, Int> = mutableMapOf()
        val regex = Regex("\\W+")
        val words = paragraph.lowercase().split(regex)
        
        for (word in words) {
            if (word.isBlank()) continue;
            if (!banned.contains(word)) {
                map[word] = map.getOrDefault(word, 0) + 1
            }
        }
        
        return map.maxBy { it.value }.key
    }
}